adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpserver/TcpServerAdapterSpace.java

```diff
@@ -0,0 +1,291 @@
+                    .setDescription("this is the name of the output file. If 'stdout', output is sent to tcpserver, not a file.")
+            )
+            .add(
+                Param.defaultTo("host","localhost")
+                    .setDescription("this is the name of the output file. If 'stdout', output is sent to tcpserver, not a file.")
+            )
+            .add(
+                Param.defaultTo("port",8080)
+                    .setDescription("this is the name of the output file. If 'stdout', output is sent to tcpserver, not a file.")
+            )
+            .add(
+                Param.defaultTo("filename","tcpserver")
+                    .setDescription("this is the name of the output file. If 'stdout', output is sent to tcpserver, not a file.")
+            )
+            .add(
+                Param.defaultTo("newline",true)
+                    .setDescription("whether to automatically add a missing newline to the end of any output\n")
+            )
+            .add(
+                Param.optional("format")
+                    .setRegex("csv|readout|json|inlinejson|assignments|diag")
+                    .setDescription("Which format to use.\n" +
+                        "If provided, the format will override any statement formats provided by the YAML. " +
+                        "If 'diag' is used, a diagnostic readout will be provided for binding constructions.")
+            )
+            .add(
+                Param.defaultTo("bindings","doc")
+                    .setDescription("This is a simple way to specify a filter for the names of bindings that you want to use.\n" +
+                        "If this is 'doc', then all the document level bindings are used. If it is any other value, it is taken\n" +
+                        "as a pattern (regex) to subselect a set of bindings by name. You can simply use the name of a binding\n" +
+                        "here as well.")
+
+            )
+            .asReadOnly();
+    }
+    private interface Shutdown {
+        void shutdown();
+    }
+    public static class SocketWriter implements Runnable, Shutdown {
+        private final BlockingQueue<String> sourceQueue;
+        private final OutputStream outputStream;
+        private final OutputStreamWriter writer;
+        private boolean running = true;
+
+
+        public SocketWriter(BlockingQueue<String> sourceQueue, Socket connectedSocket) {
+            this.sourceQueue = sourceQueue;
+            try {
+                outputStream = connectedSocket.getOutputStream();
+                this.writer = new OutputStreamWriter(outputStream);
+                //connectedSocket.shutdownInput();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        public void shutdown() {
+            this.running = false;
+        }
+
+        @Override
+        public void run() {
+            try (Writer writer = this.writer) {
```

Comment:
Maybe change the name of this `writer` variable to be something else like runWriter or threadWriter to avoid confusion with the instance variable declared above.